sched/irq/irq_csection.c

```diff
@@ -275,7 +292,7 @@ irqstate_t enter_critical_section(void)
 
   /* Check if we were called from an interrupt handler */
 
-  if (!up_interrupt_context())
+  if (!csection_degraded() && !up_interrupt_context())
```

Comment:
Maybe for style consistently you could use the dummy-if in the non-SMP case as well. This is fine by me too.